gstd/gstd_action.c

```diff
@@ -0,0 +1,240 @@
+  object_class->get_property = gstd_action_get_property;
+  object_class->dispose = gstd_action_dispose;
+
+  properties[PROP_TARGET] =
+      g_param_spec_object ("target",
+      "Target",
+      "The target object owning the action",
+      G_TYPE_OBJECT,
+      G_PARAM_READWRITE |
+      G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS | GSTD_PARAM_READ);
+
+
+  g_object_class_install_properties (object_class, N_PROPERTIES, properties);
+
+  gstdc->to_string = GST_DEBUG_FUNCPTR (gstd_action_to_string);
+  gstdc->create = GST_DEBUG_FUNCPTR (gstd_action_create_default);
+
+  /* Initialize debug category with nice colors */
+  debug_color = GST_DEBUG_FG_BLACK | GST_DEBUG_BOLD | GST_DEBUG_BG_WHITE;
+  GST_DEBUG_CATEGORY_INIT (gstd_action_debug, "gstdaction", debug_color,
+      "Gstd Action category");
+
+}
+
+static void
+gstd_action_init (GstdAction * self)
+{
+  self->target = DEFAULT_PROP_TARGET;
+}
+
+static void
+gstd_action_dispose (GObject * object)
+{
+  GstdAction *self = GSTD_ACTION (object);
+
+  if (self->target) {
+    g_object_unref (self->target);
+    self->target = NULL;
+  }
+
+  G_OBJECT_CLASS (gstd_action_parent_class)->dispose (object);
+}
+
+static void
+gstd_action_get_property (GObject * object,
+    guint property_id, GValue * value, GParamSpec * pspec)
+{
+  GstdAction *self = GSTD_ACTION (object);
+
+  switch (property_id) {
+    case PROP_TARGET:
+      GST_DEBUG_OBJECT (self, "Returning action owner %p (%s)", self->target,
+          GST_OBJECT_NAME (self->target));
+      g_value_set_object (value, self->target);
+      break;
+    default:
+      /* We don't have any other action... */
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
+      break;
+  }
+}
+
+static void
+gstd_action_set_property (GObject * object,
+    guint property_id, const GValue * value, GParamSpec * pspec)
+{
+  GstdAction *self = GSTD_ACTION (object);
+
+  switch (property_id) {
+    case PROP_TARGET:
+      if (self->target)
+        g_object_unref (self->target);
+      self->target = g_value_dup_object (value);
+      GST_DEBUG_OBJECT (self, "Setting action owner %p (%s)", self->target,
+          GST_OBJECT_NAME (self->target));
+      break;
+    default:
+      /* We don't have any other action... */
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
+      break;
+  }
+}
+
+static GstdReturnCode
+gstd_action_to_string (GstdObject * obj, gchar ** outstring)
+{
+  guint action_id;
+  GSignalQuery *query = NULL;
+  GstdAction *self;
+  const gchar *typename;
+  GstdIFormatter *formatter = NULL;
+  int i;
+
+  g_return_val_if_fail (GSTD_IS_OBJECT (obj), GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (outstring, GSTD_NULL_ARGUMENT);
+
+  self = GSTD_ACTION (obj);
+  formatter = g_object_new (obj->formatter_factory, NULL);
+
+  action_id =
+      g_signal_lookup (GSTD_OBJECT_NAME (self), G_OBJECT_TYPE (self->target));
+  query = g_new0 (GSignalQuery, 1);
```

Comment:
I suggest you make a static allocation here. For example:
```c
GSignalQuery query;

g_signal_query (action_id, &query);
```

In general, you want to avoid using pointers if possible because of their multiple problems (segfaults, leaks, etc...)